src/messages/manifestjson.js

```diff
@@ -107,6 +124,26 @@ export const MANIFEST_PERMISSIONS = {
   file: MANIFEST_JSON,
 };
 
+export const MANIFEST_PERMISSIONS_PRIVILEGED =
+  'MANIFEST_PERMISSIONS_PRIVILEGED';
+export function manifestPermissionsPrivileged(error) {
+  const messageTmpl =
+    i18n._(oneLine`%(instancePath)s: the following privileged permissions
+                     are only allowed in add-ons signed as privileged:
+                     %(privilegedPermissions)s.`);
+  const message = i18n.sprintf(messageTmpl, {
+    instancePath: error.instancePath,
+    privilegedPermissions: JSON.stringify(error.params.privilegedPermissions),
+  });
+
+  return {
+    code: MANIFEST_PERMISSIONS_PRIVILEGED,
+    message,
+    description: message,
```

Comment:
I think the `description` should include a bit more information so I'd reuse `message` + the following:

```
Please refer to https://github.com/mozilla-extensions/xpi-manifest to learn more about privileged extensions and signing.
```

Reply:
Composing the string would be technically possible by using i18n.sprintf, but Instead of repeating the message in a description defined as a composed string I oped to set the description to the string you suggested (and the message as is).

Same for the one that follows.

(As a side note, this should be the only deviation from the suggested edits, everything else should be applied exactly as in the related suggested change).

Reply:
ah ok yeah, that's probably fine. I don't know if this is consistent with what we're doing already but it isn't wrong so thank you for that